### PR TITLE
Upgrade syn and quote, rename component attrib to storage

### PR DESCRIPTION
**Pull request by torkleyy**
Not meant to merge yet.

---
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/343)

### CODE REVIEW

File: specs-derive/src/lib.rs [hunk 1]
```diff
@@ -6,61 +6,59 @@
...
-            _ => None,
-        })
-        .and_then(|attr| match *attr {
-            MetaItem::Word(ref ident) => Some(ident),
-            _ => None,
-        })
-        .cloned()
-        .unwrap_or(Ident::new("::specs::storage::DenseVecStorage"));
+        .map(|attr| attr.tts.clone())
+        //.unwrap_or_else(|| quote!("(DenseVecStorage)").into());
```
Comment (torkleyy) [inline specs-derive/src/lib.rs]: @dtolnay What's the recommended way for providing a default path here?

Comment (dtolnay) [reply specs-derive/src/lib.rs]: Maybe something like this:

```rust
let storage = ast.attrs.iter()
    .find(|attr| attr.path.segments[0].ident == "storage")
    .map(|attr| syn::parse2::<StorageAttribute>(attr.tts.clone()).unwrap().storage)
    .unwrap_or_else(|| parse_quote!(DenseVecStorage));
```

Comment (torkleyy) [reply specs-derive/src/lib.rs]: Thank you!

**Comment by torkleyy** [comment]
Thanks @Rhuagh 

bors r+

**Pull request merged**